fauna/client/client.py

```diff
@@ -48,6 +51,19 @@ class QueryOptions:
   additional_headers: Optional[Dict[str, str]] = None
 
 
+@dataclass
+class StreamOptions:
+  """
+    A dataclass representing options available for a stream.
+
+    * idle_timeout - Controls the maximum amount of time the driver will wait on stream idling.
+    * retry_on_timeout - If true, streaming is reconnected on timeouts.
+    """
+
+  idle_timeout: Optional[timedelta] = None
+  retry_on_timeout: bool = True
```

Comment:
If we do keep these, both timeout and idle timeout are overloaded in the http world, so we should be clear. In the doc string, make sure you define an idle stream.

```suggestion
  idle_stream_timeout: Optional[timedelta] = None
  retry_on_idle_stream_timeout: bool = True
```